lib/spack/spack/test/build_environment.py

```diff
@@ -209,3 +208,29 @@ def test_compiler_config_modifications(monkeypatch):
     os.environ.pop('PATH_LIST', None)
     os.environ.pop('EMPTY_PATH_LIST', None)
     os.environ.pop('NEW_PATH_LIST', None)
+
+
+@pytest.mark.regression('9107')
+def test_spack_paths_before_module_paths(config, mock_packages, monkeypatch):
+    s = spack.spec.Spec('cmake')
+    s.concretize()
+    pkg = s.package
+
+    module_path = '/path/to/module'
+
+    def _set_wrong_cc(x):
+        os.environ['PATH'] = module_path + ':' + os.environ['PATH']
+
+    monkeypatch.setattr(
+        spack.build_environment, 'load_module', _set_wrong_cc
```

Comment:
@becker33  so this must be getting executed *somewhere*, otherwise the `paths.index` call would fail below, but I don't see any modules being set for the compilers, so I'm not 100% sure this is getting exactly at what we want to test. Do you know how this is getting executed?

Reply:
@scheibelp The second monkey patch tells Spack to associate the module `some_module` with the compiler. When you try to load `some_module`, it instead calls `_set_wrong_cc` because of this monkey patch. So line 231 (`build_environment.setup_package`) calls `load_module` which calls `_set_wrong_cc`.